provider/etcd/etcd_client.go

```diff
@@ -31,12 +31,44 @@ import (
 	"github.com/prometheus/common/model"
 
 	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
 var (
 	ErrorEtcdNotInitialized     = errors.New("Etcd not initialized")
 	ErrorEtcdGetNoResult        = errors.New("etcdGet did not receive a result for fingerprint")
 	ErrorEtcdGetMultipleResults = errors.New("etcdGet received multiple results for fingerprint")
+
+	// Prometheus Counters
+	etcdCheckAndPutTotal = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "alertmanager_etcd_checkandput_total",
+			Help: "The total number of CheckAndPut calls received",
+		},
+		[]string{"status"},
+	) // "status":"filtered|accepted|error"
+	etcdOperationsTotal = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "alertmanager_etcd_operations_total",
+			Help: "The total number of operations initiated to etcd",
+		},
+		[]string{"operation", "result"},
+	) // "operation": "get|put|delete", "result":"success|error"
+	etcdWatchOperationsTotal = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "alertmanager_etcd_watch_operations_total",
+			Help: "The total number of operations received from etcd watch",
+		},
+		[]string{"operation"},
+	) // "operation":"put|delete"
+	etcdQueueLength = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "alertmanager_etcd_queue_length",
+			Help: "The total number of operations pending to be processed by etcd watch",
+		},
+		[]string{"name"},
+	)
 )
 
 const EtcdTimeoutGet = 150 * time.Millisecond
```

Comment:
MAY There are some hardcoded timeout values around, it will be helpful to make them configurable to fit into different environments without re-compile the executable.

Reply:
@mpeters Recommended this.  Its a good idea.  I suppose we should polish it.